### PR TITLE
dnsdist: Unify non-terminal actions as SetXXXAction()

### LINKED ISSUE #8118: dnsdist: Unify non-terminal actions as SetXXXAction()
- Program: dnsdist
 - Issue type: Feature request

### Short description

As suggested by @zeha in [1], we should do a refactoring of our actions and turn these that do not need to stop the processing of the subsequent rules into SetXXXAction().
With a bit of luck we might be able to deprecate the existing ones first, I did not look into it.

[1]: https://mailman.powerdns.com/pipermail/dnsdist/2019-July/000595.html

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

Closes #8118.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [x] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by rgacogne** [comment]
Rebased on main to fix a conflict.

File: pdns/dnsdist-lua-actions.cc [hunk 1]
```diff
@@ -1618,12 +1675,22 @@ void setupLuaActions(LuaContext& luaCtx)
       return std::shared_ptr<DNSAction>(new LuaFFIAction(func));
     });
 
+  luaCtx.writeFunction("SetNoRecurseAction", []() {
+      return std::shared_ptr<DNSAction>(new SetNoRecurseAction);
+    });
+
   luaCtx.writeFunction("NoRecurseAction", []() {
-      return std::shared_ptr<DNSAction>(new NoRecurseAction);
+      warnlog("access to NoRecurseAction is deprecated, please use SetNoRecurseAction instead");
```
Comment (pieterlexis) [inline pdns/dnsdist-lua-actions.cc:1683]: perhaps add "and will be removed in a future version"?

File: pdns/dnsdistdist/docs/rules-actions.rst [hunk 2]
```diff
@@ -1211,19 +1226,52 @@ The following actions exist.
...
 .. function:: NoneAction()
 
   Does nothing.
   Subsequent rules are processed after this action.
 
 .. function:: NoRecurseAction()
 
+  .. deprecated:: 1.6.0
+
+  This function has been deprecated in 1.6.0, please use :func:`NoRecurseAction` instead.
```
Comment (pieterlexis) [inline pdns/dnsdistdist/docs/rules-actions.rst:1273]:
```suggestion
  This function has been deprecated in 1.6.0, please use :func:`SetNoRecurseAction` instead.
```

File: pdns/dnsdistdist/docs/rules-actions.rst [hunk 3]
```diff
@@ -1450,6 +1619,10 @@ The following actions exist.
 
   .. versionadded:: 1.3.0
 
+   .. deprecated:: 1.6.0
```
Comment (pieterlexis) [inline pdns/dnsdistdist/docs/rules-actions.rst:1622]:
```suggestion
  .. deprecated:: 1.6.0
```

**Comment by zeha** [comment]
Nice! <3

**Pull request merged**